app/src/main/java/com/rockthevote/grommet/ui/eventFlow/PartnerLoginViewModel.kt

```diff
@@ -0,0 +1,115 @@
+package com.rockthevote.grommet.ui.eventFlow
+
+import androidx.lifecycle.*
+import com.rockthevote.grommet.data.api.RockyService
+import com.rockthevote.grommet.data.db.dao.PartnerInfoDao
+import com.rockthevote.grommet.data.db.model.PartnerInfo
+import com.rockthevote.grommet.util.coroutines.DispatcherProvider
+import com.rockthevote.grommet.util.coroutines.DispatcherProviderImpl
+import kotlinx.coroutines.CoroutineExceptionHandler
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.SupervisorJob
+import kotlinx.coroutines.launch
+import timber.log.Timber
+
+/**
+ * Created by Mechanical Man on 5/16/20.
+ */
+class PartnerLoginViewModel(
+        dispatchers: DispatcherProvider = DispatcherProviderImpl(),
+        private val rockyService: RockyService,
+        private val partnerInfoDao: PartnerInfoDao
+) : ViewModel() {
+
+    val partnerInfoId: LiveData<Long> =
+            Transformations.map(partnerInfoDao.getCurrentPartnerInfo()) { result ->
+                result.partnerInfoId
+            }
+
+    private val _partnerLoginState = MutableLiveData<PartnerLoginState>(PartnerLoginState.Init)
+    val partnerLoginState: LiveData<PartnerLoginState> = _partnerLoginState
+
+    private val _effect = MutableLiveData<PartnerLoginState.Effect?>()
+    val effect: LiveData<PartnerLoginState.Effect?> = _effect
+
+    private val coroutineExceptionHandler = CoroutineExceptionHandler { _, throwable ->
+        // TODO Should we handle this another way? Is it needed?
+        Timber.e(throwable)
+    }
+
+    private val supervisorJob = SupervisorJob()
+
+    private val rockyRequestScope = CoroutineScope(dispatchers.io + coroutineExceptionHandler)
+
+    init {
+        viewModelScope.launch(dispatchers.io) {
+            updateState(PartnerLoginState.Init)
+        }
```

Comment:
You're already initializing the MutableLiveData with `Init`, you don't need to do this here.

Reply:
👍